### PR TITLE
Issue 1083 pdb0atoms

### LINKED ISSUE #1083: Can't write PDB with 0 atoms
### Expected behaviour

When I have an atomgroup with 0 atoms I should get an error that this PDB can't be written due to that there are no atoms in the atomgroup

### Actual behaviour

I get an internal error that I'm calling `_write_pdb_title` gets an unexpected number of arguments

### Code to reproduce the behaviour

``` python
import MDAnalysis as mda

u = mda.Universe(top, trj)
ag = u.select_atoms('resid 100000')

with mda.Writer('test.pdb', ag.n_atoms) as w:
    w.write(ag)

```

### Currently version of MDAnalysis:
(run `python -c "import MDAnalysis as mda; print(mda.__version__)"`)
Develop version

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #1083

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/coordinates/PDB.py [hunk 1]
```diff
@@ -488,40 +492,35 @@ class PDBWriter(base.Writer):
...
            comments to annotate pdb file (added to the TITLE record); note that
            any remarks from the trajectory that serves as input are
            written to REMARK records with lines longer than :attr:`remark_max_length` (66
            characters) being wrapped.
-         *convert_units*
+        convert_units
            units are converted to the MDAnalysis base format; ``None`` selects
            the value of :data:`MDAnalysis.core.flags` ['convert_lengths']
-         *bonds*
+        bonds
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/PDB.py]: thanks for the docs

File: package/MDAnalysis/coordinates/PDB.py [hunk 2]
```diff
@@ -488,40 +492,35 @@ class PDBWriter(base.Writer):
...
-
-         *multiframe*
+        multiframe : bool, optional
            ``False``: write a single frame to the file; ``True``: create a
            multi frame PDB file in which frames are written as MODEL_ ... ENDMDL_
            records. If ``None``, then the class default is chosen.    [``None``]
+        keep_atomids : bool, optional
+           If ``True`` the atom indices as defined in the AtomGroup are written to
+           the output, otherwise atom indices will be numbered sequentially from 1.
+           [``False``]
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/PDB.py]: What is the purpose of the last line with False?

File: package/MDAnalysis/coordinates/PDB.py [hunk 3]
```diff
@@ -574,9 +578,6 @@ def _write_pdb_title(self):
                        "".format(self.start, self.remarks))
 
     def _write_pdb_header(self):
-        if not self.obj or not hasattr(self.obj, 'universe'):
-            self._write_pdb_title(self)
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/PDB.py]: doesn't this need to be called anymore?

File: package/MDAnalysis/coordinates/PDB.py (hunk 2 above)
Comment (richardjgowers) [inline package/MDAnalysis/coordinates/PDB.py]: Default value

File: package/MDAnalysis/coordinates/PDB.py (hunk 3 above)
Comment (richardjgowers) [inline package/MDAnalysis/coordinates/PDB.py]: There's already a check before this method is called (`_update_frame`) so this check was redundant.

**Comment by richardjgowers** [comment]
@kain88-de I've rolled this back so it just deals with one issue, the other issues will have to wait

**Comment by kain88-de** [comment]
What other issues where solved in this branch?

**Comment by richardjgowers** [comment]
I did try and solve the atom renumbering, but the pdbwriter is a bit of a mess and I got stuck

**Comment by richardjgowers** [comment]
Which is #1072

File: package/MDAnalysis/coordinates/PDB.py [hunk 4]
```diff
@@ -638,67 +636,40 @@ def _check_pdb_coordinates(self):
...
 
-        atoms = sorted([a.index for a in self.obj.atoms])
+        atoms = np.sort(self.obj.atoms.indices)
 
-        conect = [([a, ] + sorted(con[a])) for a in atoms if a in con]
+        conect = ([a] + sorted(con[a])
+                  for a in atoms if a in con)
 
-        conect = [[mapping[e] for e in row] for row in conect]
+        self.remap_indices = True
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/PDB.py]: Remapping stuff is still here

Comment (richardjgowers) [inline package/MDAnalysis/coordinates/PDB.py]: fixed

**Comment by kain88-de** [comment]
CHANGELOG is missing otherwise it's ready to merge.

**Comment by richardjgowers** [comment]
Ok done

**Comment by kain88-de** [comment]
Thanks. You can merge it yourself when travis has finished.

**Pull request merged**